Code/Engine/GameEngine/Gameplay/Implementation/RaycastComponent.cpp

```diff
@@ -0,0 +1,190 @@
+
+#include <GameEnginePCH.h>
+
+#include <GameEngine/Gameplay/RaycastComponent.h>
+#include <GameEngine/Interfaces/PhysicsWorldModule.h>
+#include <Core/WorldSerializer/WorldReader.h>
+#include <Core/WorldSerializer/WorldWriter.h>
+#include <Core/Messages/TriggerMessage.h>
+#include <RendererCore/Debug/DebugRenderer.h>
+
+// clang-format off
+EZ_BEGIN_COMPONENT_TYPE(ezRaycastComponent, 1, ezComponentMode::Dynamic)
```

Comment:
The component itself is not dynamic, is it? It only changes the transform of the target.

Reply:
Yeah, that is a thing I thought about. I have a change which makes this static but adds code which makes the target object dynamic if it isn't.

Reply:
I think that can be left to the user, it will print a warning if you forget.